### ISSUE TITLE
Save `initial` and `latest` models in addition to `best`

### ISSUE THREAD

**Issue by sarthakpati** [mlcommons/GaNDLF#538]
**Is your feature request related to a problem? Please describe.**
In a lot of cases, the `latest` model might be more informative WRT how the training loss has been behaving as compared to the `latest`. Also, the `initial` model can give greater insight into post-training data.

**Describe the solution you'd like**
In addition to the `best` model, save `initial` (i.e., the model right before any training has occurred) and `latest` models.

**Describe alternatives you've considered**
N.A.

**Additional context**
N.A.
